server/routes/bookingJourney/selectVisitDateTimeController.ts

```diff
@@ -0,0 +1,37 @@
+import type { RequestHandler } from 'express'
+import { VisitSessionsService } from '../../services'
+
+export default class SelectVisitDateTimeController {
+  public constructor(private readonly visitSessionsService: VisitSessionsService) {}
+
+  public view(): RequestHandler {
+    return async (req, res) => {
+      const { prison, prisoner, selectedVisitors } = req.session.bookingJourney
+
+      const selectedVisitorIds = selectedVisitors.map(visitor => visitor.personId)
+
+      const { calendar, firstSessionDate } = await this.visitSessionsService.getVisitSessionsCalendar(
+        prisoner.prisonCode,
+        prisoner.prisonerNumber,
+        selectedVisitorIds,
+        prison.policyNoticeDaysMax,
+      )
+
+      res.render('pages/bookingJourney/selectVisitDateTime', {
+        calendar,
+        selectedDate: firstSessionDate,
+        prisoner,
+      })
+    }
+  }
+
+  public submit(): RequestHandler {
+    return async (req, res) => {
+      const { visitSession } = req.body
+
+      console.log('Selected visit session: ', visitSession)
```

Comment:
Next to do is validate the requested session, create a visit application and store to`req.session`...